core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -719,6 +778,103 @@ private ExprValue exprWeek(ExprValue date, ExprValue mode) {
+  }
+
+  private ExprValue unixTimeStampOf(ExprValue value) {
+    var res = unixTimeStampOfImpl(value);
+    if (res == null) {
+      return ExprNullValue.of();
+    }
+    if (res < 0) {
+      // According to MySQL returns 0 if year < 1970, don't return negative values as java does.
+      return new ExprDoubleValue(0);
+    }
+    if (res >= 32536771200d) {
+      // Return 0 also for dates > '3001-01-19 03:14:07.999999' UTC (32536771199.999999 sec)
+      return new ExprDoubleValue(0);
+    }
+    return new ExprDoubleValue(res);
+  }
+
+  private Double unixTimeStampOfImpl(ExprValue value) {
+    // Also, according to MySQL documentation:
+    //    The date argument may be a DATE, DATETIME, or TIMESTAMP ...
+    switch ((ExprCoreType)value.type()) {
+      case DATE: return value.dateValue().toEpochSecond(LocalTime.MIN, ZoneOffset.UTC) + 0d;
+      case DATETIME: return value.datetimeValue().toEpochSecond(ZoneOffset.UTC)
+          + value.datetimeValue().getNano() / 1E9;
+      case TIMESTAMP: return value.timestampValue().getEpochSecond()
+          + value.timestampValue().getNano() / 1E9;
+      default:
+        //     ... or a number in YYMMDD, YYMMDDhhmmss, YYYYMMDD, or YYYYMMDDhhmmss format.
+        //     If the argument includes a time part, it may optionally include a fractional
+        //     seconds part.
+
+        var dateFormatShortYear = new DateTimeFormatterBuilder()
+            .appendValueReduced(YEAR, 2, 2, 1970)
+            .appendPattern("MMdd")
+            .toFormatter()
+            .withResolverStyle(ResolverStyle.STRICT);
+
+        var dateFormatLongYear = new DateTimeFormatterBuilder()
+            .appendValue(YEAR, 4)
+            .appendPattern("MMdd")
+            .toFormatter()
+            .withResolverStyle(ResolverStyle.STRICT);
+
+        var dateTimeFormatShortYear = new DateTimeFormatterBuilder()
+            .appendValueReduced(YEAR, 2, 2, 1970)
+            .appendPattern("MMddHHmmss")
+            .toFormatter()
+            .withResolverStyle(ResolverStyle.STRICT);
+
+        var dateTimeFormatLongYear = new DateTimeFormatterBuilder()
+            .appendValue(YEAR,4)
+            .appendPattern("MMddHHmmss")
+            .toFormatter()
+            .withResolverStyle(ResolverStyle.STRICT);
+
+        var format = new DecimalFormat("0.#");
+        format.setMinimumFractionDigits(0);
+        format.setMaximumFractionDigits(6);
+        String input = format.format(value.doubleValue());
+        double fraction = 0;
+        if (input.contains(".")) {
+          // Keeping fraction second part and adding it to the result, don't parse it
+          // Because `toEpochSecond` returns only `long`
+          // input = 12345.6789 becomes input = 12345 and fraction = 0.6789
+          fraction = value.doubleValue() - Math.round(Math.ceil(value.doubleValue()));
+          input = input.substring(0, input.indexOf('.'));
+        }
+        try {
```

Comment:
I was thinking there should be a better way to do this, like https://stackoverflow.com/questions/36188428/java-8-date-equivalent-to-jodas-datetimeformatterbuilder-with-multiple-parser-f
But your return values are different based on the format.  So maybe this is best.

Reply:
Unfortunately, it doesn't work. I tried
```java
new DateTimeFormatterBuilder()
            .optionalStart()
            .appendValueReduced(YEAR, 2, 2, 1970)
            .appendPattern("MMdd")
            .optionalEnd()
            .optionalStart()
            .appendValue(YEAR, 4)
            .appendPattern("MMdd")
            .optionalEnd()
            .toFormatter()
            .withResolverStyle(ResolverStyle.STRICT);
```
and
```java
new DateTimeFormatterBuilder()
            .optionalStart()
            .appendValueReduced(YEAR, 2, 2, 1970)
            .optionalEnd()
            .optionalStart()
            .appendValue(YEAR, 4)
            .optionalEnd()
            .appendPattern("MMdd")
            .toFormatter()
            .withResolverStyle(ResolverStyle.STRICT);
```